### PR TITLE
[Feature] Profile Category

### PR DESCRIPTION
**Pull request by gertie-sheshe**
## Description

Add Profile Category component

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Screenshots
Screenshot 2021-09-14 at 10 36 28
Screenshot 2021-09-14 at 10 36 57

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/components/HURUmap/CategoryHeader/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import Image from "next/image";
+import PropTypes from "prop-types";
+import React from "react";
+
+import useStyles from "./useStyles";
+
+const CategoryHeader = ({ title, description, image }) => {
+  const classes = useStyles();
+  return (
+    <div>
```
Comment (kilemensi) [inline src/components/HURUmap/CategoryHeader/index.js:11]:
```suggestion
    <div className={classes.root}>
```

File: src/components/HURUmap/CategoryHeader/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          <div className={classes.image}>
+            <Image src={image} layout="fill" />
+          </div>
+        )}
+        {title && (
+          <Typography variant="h3" className={classes.title}>
+            {title}
+          </Typography>
+        )}
+      </Grid>
```
Comment (kilemensi) [inline src/components/HURUmap/CategoryHeader/index.js:23]:
```suggestion
      <Typography variant="h3" className={classes.title}>
        <div className={classes.image}>
          <Image src={image} layout="fill" />
        </div>
        {title}
      </Typography>
```

We can then set `display: "flex", alignItems: "center" in useStyles `to achieve the same effect without using `Grid`.

File: src/components/HURUmap/CategoryHeader/useStyles.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  title: {
+    fontSize: typography.pxToRem(28),
+  },
+  image: {
+    position: "relative",
+    height: typography.pxToRem(55.5),
+    width: typography.pxToRem(55.5),
+  },
+  description: {
+    fontWeight: "400",
```
Comment (kilemensi) [inline src/components/HURUmap/CategoryHeader/useStyles.js:14]: Isn't it supposed to be Medium i.e. `500` (which is `body2` in the theme).

**Review by kilemensi — APPROVED** [review]
🚀 

----

Not that important but is it possible to use `Icon` instead of `Image`?

File: src/components/HURUmap/CategoryHeader/index.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        </Typography>
+      )}
+    </div>
+  );
+};
+
+CategoryHeader.propTypes = {
+  title: PropTypes.string,
+  description: PropTypes.string,
+  image: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/HURUmap/CategoryHeader/index.js]: Fixed

File: src/components/HURUmap/CategoryHeader/useStyles.js (hunk 3 above)
Comment (gertie-sheshe) [reply src/components/HURUmap/CategoryHeader/useStyles.js]: Fixed

File: src/components/HURUmap/CategoryHeader/index.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import { Typography, Grid } from "@material-ui/core";
+import Image from "next/image";
+import PropTypes from "prop-types";
+import React from "react";
+
+import useStyles from "./useStyles";
+
+const CategoryHeader = ({ title, description, image }) => {
+  const classes = useStyles();
+  return (
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/HURUmap/CategoryHeader/index.js:15]: Fixed

File: src/components/HURUmap/CategoryHeader/index.js (hunk 1 above)
Comment (gertie-sheshe) [reply src/components/HURUmap/CategoryHeader/index.js]: Fixed

File: src/components/HURUmap/CategoryHeader/index.js (hunk 2 above)
Comment (gertie-sheshe) [reply src/components/HURUmap/CategoryHeader/index.js]: Fixed

File: src/components/HURUmap/CategoryHeader/index.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import useStyles from "./useStyles";
+
+const CategoryHeader = ({ title, description, icon }) => {
+  const classes = useStyles();
+
+  if (!title) {
+    return null;
+  }
+
+  return (
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/HURUmap/CategoryHeader/index.js]: Fixed

File: src/components/HURUmap/CategoryHeader/index.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    <div className={classes.root}>
+      <Typography variant="h3" className={classes.title}>
+        <div className={classes.icon}>
+          <Icon src={icon} layout="fill" />
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/HURUmap/CategoryHeader/index.js]: Oooh 😅 

Fixed

File: src/components/HURUmap/CategoryHeader/index.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+  if (!title) {
+    return null;
+  }
+  return (
+    <div className={classes.root}>
+      <Typography variant="h3" className={classes.title}>
+        <SvgIcon
+          className={classes.svgIcon}
+          component={icon}
```
Comment (kilemensi) [inline src/components/HURUmap/CategoryHeader/index.js:18]: To confirm, does this work with icon links with get from CMS? e.g. does a link like <https://cms.dev.codeforafrica.org/pesayetu/wp-content/uploads/sites/2/2021/08/Group-4658-1.svg> work?

File: src/components/HURUmap/CategoryHeader/useStyles.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import { makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles(({ typography }) => ({
+  root: {},
+  title: {
+    fontSize: typography.pxToRem(28),
+    display: "flex",
+    alignItems: "center",
+  },
+
```
Comment (kilemensi) [inline src/components/HURUmap/CategoryHeader/useStyles.js:10]:
```suggestion
```

**Review by kilemensi — COMMENTED** [review]
👍🏾

File: package.json [hunk 10]
```diff
@@ -28,6 +28,7 @@
     "@material-ui/core": "^4.12.1",
     "@material-ui/lab": "^4.0.0-alpha.60",
     "@material-ui/utils": "^4.11.2",
+    "@mui/material": "^5.0.0-rc.0",
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply package.json]: Hehehe...where? I don't see anything @kilemensi

File: src/components/HURUmap/CategoryHeader/useStyles.js (hunk 9 above)
Comment (gertie-sheshe) [reply src/components/HURUmap/CategoryHeader/useStyles.js]: Fixed

File: src/components/HURUmap/CategoryHeader/index.js [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+          viewBox="0 0 130 130"
+        />
+        {title}
+      </Typography>
+      <Typography variant="body2" className={classes.description}>
+        {description}
+      </Typography>
```
Comment (KhadijaMahanga) [inline src/components/HURUmap/CategoryHeader/index.js:25]: @gertie-sheshe description comes as an html. We should use RichTypography

**Review by KhadijaMahanga — APPROVED** [review]

File: src/components/HURUmap/CategoryHeader/index.js (hunk 11 above)
Comment (gertie-sheshe) [reply src/components/HURUmap/CategoryHeader/index.js]: Fixed

File: src/components/HURUmap/CategoryHeader/index.js (hunk 8 above)
Comment (gertie-sheshe) [reply src/components/HURUmap/CategoryHeader/index.js:19]: This does not work, an element type is needed to be passed to the SvgIcon component @kilemensi I propose switching it back to next/images and look into It in a different PR as everything else is ready in this PR

Comment (kilemensi) [reply src/components/HURUmap/CategoryHeader/index.js:19]: Yeah was afraid of that @gertie-sheshe ... We'll probably need something like https://stackoverflow.com/a/52967417 but it feels like too much work for something so small. So yes, lets just render SVG using `Image` component.

Comment (gertie-sheshe) [reply src/components/HURUmap/CategoryHeader/index.js:19]: Thanks. Was looking at that exact link with @KobbyMmo yesterday too.

**Pull request merged**